File: L0/cpp/app/main.cpp
#include <primes.hpp>
#include <valueConverter.hpp>

#include <iostream>


int main(int argc, const char* const argv[])
{
    for (size_t i = 1; i < static_cast<size_t>(argc); ++i)
    {
        try
        {
            const int argNumber = ValueConverter::stringToVal<int>(std::string(argv[i]));
            const int gDiv = Primes::gDiv(argNumber);

            std::cout << std::to_string(argNumber) << " --> " << std::to_string(gDiv) << std::endl;
        }
        catch (const std::invalid_argument &e)
        {
            std::cerr << "Error: " << e.what() << std::endl;
        }
    }

    return 0;
}

File: L0/java/Main.java
public class Main
{
    private Main() throws InstantiationError
    {
        throw new InstantiationError("This is a static class!");
    }

    public static void main(final String[] args)
    {
        for (final String arg : args)
        {
            try
            {
                final int argNumber = Integer.parseInt(arg);
                final int gDiv = Primes.gDiv(argNumber);
                System.out.println(argNumber + " --> " + gDiv);
            }
            catch (final IllegalArgumentException e) // NumberFormatException is a subclass of alternative IllegalArgumentException
            {
                System.out.println(e.toString());
            }
        }
    }
}
